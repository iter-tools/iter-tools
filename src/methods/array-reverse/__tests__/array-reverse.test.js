import { arrayReverse, toArray } from '../../..';

describe('arrayReverse', () => {
  describe('when source is empty', () => {
    it('yields nothing', () => {
      expect(toArray(arrayReverse(null))).toEqual([]);
      expect(toArray(arrayReverse(undefined))).toEqual([]);
      expect(toArray(arrayReverse([]))).toEqual([]);
    });
  });

  describe('when source is an array', () => {
    it('yields array values in reverse order', () => {
      expect(toArray(arrayReverse([1, 2, 3]))).toEqual([3, 2, 1]);
    });
  });

  describe('when source is a string', () => {
    it('yields characters in reverse order', () => {
      expect(toArray(arrayReverse('abc'))).toEqual(['c', 'b', 'a']);
    });
  });

  describe('when input is invalid', () => {
    const source: any = 0;
    expect(() => toArray(arrayReverse(source))).toThrowErrorMatchingSnapshot();
  });
});
