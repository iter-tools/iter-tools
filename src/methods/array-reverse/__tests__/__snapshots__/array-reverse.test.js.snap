// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"arrayReverse only works arrays and strings"`;
